### PR TITLE
Emit events for inflation disable and staking rewards

### PR DESCRIPTION
**Pull request by drklee3**
## Description
- Emits event when inflation is disabled
- Emits event on each staking reward payout

## Checklist
 - [ ] Changelog has been updated as necessary.

### CODE REVIEW

File: app/test_common.go [hunk 1]
```diff
@@ -536,3 +537,31 @@ func NewFundedGenStateWithSameCoinsWithModuleAccount(cdc codec.JSONCodec, coins
...
+
+func attrsToMap(attrs []abci.EventAttribute) []sdk.Attribute { // new cosmos changed the event attribute type
+	out := []sdk.Attribute{}
+
+	for _, attr := range attrs {
+		out = append(out, sdk.NewAttribute(string(attr.Key), string(attr.Value)))
+	}
+
+	return out
+}
```
Comment (drklee3) [inline app/test_common.go:567]: copied from https://github.com/Kava-Labs/kava/pull/1729

File: x/community/keeper/staking.go [hunk 2]
```diff
@@ -61,7 +61,13 @@ func (k Keeper) PayoutAccumulatedStakingRewards(ctx sdk.Context) {
 	// save state
 	k.SetStakingRewardsState(ctx, state)
 
-	return
+	// emit event
+	ctx.EventManager().EmitEvent(
```
Comment (drklee3) [inline x/community/keeper/staking.go:65]: do we want to still emit events even if there are no payouts? or only when truncatedRewards is non-zero

Comment (nddeluca) [reply x/community/keeper/staking.go:65]: I'm split but leaning towards only emit if non-zero and we can move this into the if statement above.

Comment (nddeluca) [reply x/community/keeper/staking.go:65]: On deeper thought, ya let's only emit when we actually transfer funds

File: x/community/keeper/staking.go [hunk 3]
```diff
@@ -51,6 +51,14 @@ func (k Keeper) PayoutAccumulatedStakingRewards(ctx sdk.Context) {
 			// occur in cases where the chain is running in an invalid state
 			panic(err)
 		}
+
+		// emit event
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeStakingRewardsPaid,
+				sdk.NewAttribute(types.AttributeKeyStakingRewardAmount, sdk.NewCoin(stakingRewardDenom, truncatedRewards).String()),
```
Comment (nddeluca) [inline x/community/keeper/staking.go]: I think we can use ` transferAmount.String())` here and get the same output?

Comment (nddeluca) [reply x/community/keeper/staking.go]: Pushed this change

File: x/community/keeper/staking.go [hunk 4]
```diff
@@ -60,8 +68,6 @@ func (k Keeper) PayoutAccumulatedStakingRewards(ctx sdk.Context) {
 
 	// save state
 	k.SetStakingRewardsState(ctx, state)
-
-	return
```
Comment (nddeluca) [inline x/community/keeper/staking.go:64]: Ah nice right don't need explicit return

File: x/community/testutil/staking_rewards.go [hunk 5]
```diff
@@ -258,10 +258,15 @@ func (suite *stakingRewardsTestSuite) TestStakingRewards() {
...
 			keeper.SetParams(ctx, params)
 
+			stakingRewardEvents := sdk.Events{}
+
 			for {
 				// run community begin blocker logic
 				suite.testFunc(ctx, keeper)
 
+				// accumulate event rewards from events
+				stakingRewardEvents = append(stakingRewardEvents, filterStakingRewardEvents(ctx.EventManager().Events())...)
```
Comment (nddeluca) [inline x/community/testutil/staking_rewards.go:268]: I made this a little cleaner by moving the event accumulation above the break statement to catch all testFunc runs, in addition collected the events so we can add the assertion for no tracking for zero rewards

File: x/community/testutil/staking_rewards.go [hunk 6]
```diff
@@ -287,6 +292,20 @@ func (suite *stakingRewardsTestSuite) TestStakingRewards() {
...
 			suite.Equal(tc.expectedRewardsTotal.String(), feeCollectorBalanceAdded.String(), "expected fee collector balance to match")
 
+			if tc.expectedRewardsTotal.IsZero() {
+				suite.Equal(0, len(stakingRewardEvents), "expected no events to be emitted")
+			} else {
+				// we add up all reward coin events
+				eventCoins := getRewardCoinsFromEvents(stakingRewardEvents)
+
+				// assert events emitted match expected rewards
+				suite.Equal(
```
Comment (nddeluca) [inline x/community/testutil/staking_rewards.go:302]: I really like this method of asserting event balances match 👍

**Review by nddeluca — APPROVED** [review]

**Pull request merged**